rewrite-maven/src/main/java/org/openrewrite/maven/IdentifyUnreachableRepos.java

```diff
@@ -0,0 +1,155 @@
+package org.openrewrite.maven;
+
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import lombok.AllArgsConstructor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.HttpSenderExecutionContextView;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.NonNull;
+import org.openrewrite.ipc.http.HttpSender;
+import org.openrewrite.xml.XPathMatcher;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.*;
+
+public class IdentifyUnreachableRepos extends Recipe {
+    private static final XPathMatcher REPOSITORY_URL_MATCHER = new XPathMatcher("/project/repositories/repository/url");
+    private static final XPathMatcher PLUGIN_REPOSITORY_URL_MATCHER = new XPathMatcher("/project/pluginRepositories/pluginRepository/url");
+    private static final XPathMatcher DISTRIBUTION_MANAGEMENT_REPOSITORY_URL_MATCHER = new XPathMatcher("/project/distributionManagement/repository/url");
+    private static final XPathMatcher DISTRIBUTION_MANAGEMENT_SNAPSHOT_REPOSITORY_URL_MATCHER = new XPathMatcher("/project/distributionManagement/snapshotRepository/url");
+
+    public static final Map<String, String> KNOWN_DEFUNCT;
+    public static final Set<String> KNOWN_ACTIVE;
+
+    static {
+        // KNOWN_DEFUNCT is a growing list containing repositories that are known to be defunct
+        Map<String, String> knownDefunct = new HashMap<>();
+        knownDefunct.put("maven.glassfish.org/content/groups/public", "maven.java.net");
+        knownDefunct.put("abcmaven.com", null);
+        KNOWN_DEFUNCT = Collections.unmodifiableMap(knownDefunct);
+
+        //KNOWN_ACTIVE is a (growing) list containing repositories that are known to always be active
+        Set<String> knownNonDefunct = new HashSet<>();
+        KNOWN_ACTIVE = Collections.unmodifiableSet(knownNonDefunct);
```

Comment:
We can go ahead and add Maven Central and nexus snapshots repos to this... no point in pinging them many times as we run this recipe on many repositories